lib/src/utils.dart

```diff
@@ -53,6 +56,78 @@ final dotBat = Platform.isWindows ? ".bat" : "";
 /// The `.exe` extension on Windows, the empty string everywhere else.
 final dotExe = Platform.isWindows ? ".exe" : "";
 
+/// The combined license text for the package and all its dependencies.
+///
+/// We include all dependency licenses because their code may be compiled into
+/// binary and JS releases.
+Future<String> get license => _licenseMemo.runOnce(() async {
+      // A map from license texts to the set of packages that have that same
+      // license. This allows us to de-duplicate repeated licenses, such as those
+      // from Dart Team packages.
+      var licenses = <String, List<String>>{};
+      var thisPackageLicense = _readLicense(".");
+      if (thisPackageLicense != null) {
+        licenses[thisPackageLicense] = [humanName];
+      }
+
+      licenses
+          .putIfAbsent(
+              File(p.join(sdkDir.path, 'LICENSE')).readAsStringSync(), () => [])
+          .add("Dart SDK");
+
+      // Parse the package config rather than the pubspec so we include transitive
+      // dependencies. This also includes dev dependencies, but it's possible those
+      // are compiled into the distribution anyway (especially for stuff like
+      // `node_preamble`).
+      var packageConfigUrl = await Isolate.packageConfig;
+      var packageConfig = package_config.parse(
+          File(p.fromUri(packageConfigUrl)).readAsBytesSync(),
+          packageConfigUrl);
+
+      // Sort the dependencies alphabetically to guarantee a consistent
+      // ordering.
+      var dependencies = packageConfig.keys.toList()..sort();
+      for (var package in dependencies) {
+        // Don't double-include this package's license.
+        if (package == pubspec.name) continue;
+
+        var dependencyLicense =
+            _readLicense(p.dirname(p.fromUri(packageConfig[package])));
+        if (dependencyLicense != null) {
```

Comment:
If no license is present for a dependency, this probably should output a warning to the user, since that could mean that the dependency is not licensed, and therefore not distributable.

In the future, it might make sense to have this detect common license incompatibilities (e.g. a GPL dependency for a BSD package) and warn in those cases as well, though it's probably low-priority given nearly all Dart packages are licensed permissively.